situp-plugins/apmtracesource/src/test/java/com/amazon/situp/plugins/processor/oteltrace/OTelTraceRawProcessorTest.java

```diff
@@ -0,0 +1,97 @@
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.trace.v1.Span;
+import io.opentelemetry.proto.trace.v1.Status;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+import java.nio.ByteBuffer;
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class OTelTraceRawProcessorTest {
+
+    @Test
+    public void testGetJsonFromProtobufObj() throws InvalidProtocolBufferException {
+
+        final long time = System.nanoTime();
+        ByteBuffer traceIdBuffer = ByteBuffer.allocate(2 * Long.BYTES);
+        traceIdBuffer.putLong(1);
+        traceIdBuffer.putLong(2);
+        ByteBuffer spanIdBuffer = ByteBuffer.allocate(Long.BYTES);
+        spanIdBuffer.putLong(3);
+
+        ResourceSpans rs = ResourceSpans.newBuilder()
+                .setResource(Resource.newBuilder()
+                        .addAttributes(KeyValue.newBuilder()
+                                .setKey("resource-key")
+                                .build()
+                        )
+                        .build()
+                )
+                .addInstrumentationLibrarySpans(InstrumentationLibrarySpans.newBuilder()
+                        .setInstrumentationLibrary(InstrumentationLibrary.newBuilder()
+                                .setName("test-instrumentation")
+                                .setVersion("0.1.0")
+                                .build()
+                        )
+                        .addSpans(Span.newBuilder()
+                                .setTraceId(ByteString.copyFrom(traceIdBuffer.array()))
+                                .setSpanId(ByteString.copyFrom(spanIdBuffer.array()))
+                                .setName("test-span")
+                                .setKind(Span.SpanKind.CONSUMER)
+                                .setStartTimeUnixNano(time)
+                                .setEndTimeUnixNano(time + 1L)
+                                .setStatus(Status.newBuilder().setCodeValue(Status.StatusCode.Aborted_VALUE).setMessage("status-description").build())
+                                .addAttributes(KeyValue.newBuilder()
+                                        .setKey("x")
+                                        .build()
+                                )
+                                .setDroppedAttributesCount(1)
+                                .addEvents(Span.Event.newBuilder().setName("event-1").setTimeUnixNano(time + 2).build())
+                                .addEvents(Span.Event.newBuilder().setName("event-2").setTimeUnixNano(time + 2).build())
+                                .setDroppedEventsCount(2)
+                                .build())
+                        .build()
+                )
+                .build();
+
+        Record<ResourceSpans> records = new Record<>(rs);
+        OTelTraceRawProcessor.getJsonFromProtobufObj(records);
```

Comment:
are we not asserting anything on the result?

Reply:
For now, I only verified that a record is being created, is not null. I can add more test cases.

Reply:
This is fine for now.

Reply:
Added few more imports.